packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -239,14 +239,37 @@ async function redeemSharingLink(
 				const encodedShareUrl = getEncodedShareUrl(
 					odspResolvedUrl.shareLinkInfo?.sharingLinkToRedeem,
 				);
-				const redeemUrl = `${odspResolvedUrl.siteUrl}/_api/v2.0/shares/${encodedShareUrl}`;
-				const { url, headers } = getUrlAndHeadersWithAuth(
-					redeemUrl,
-					storageToken,
-					forceAccessTokenViaAuthorizationHeader,
-				);
-				headers.prefer = "redeemSharingLink";
-				return fetchAndParseAsJSONHelper(url, { headers });
+				let redeemUrl: string | undefined;
+				// There is an issue where if we use the siteUrl in /shares, then the allowed length of url is just a few hundred characters
+				// and we fail to do the redeem. But if we use the tenant domain in the url, then the allowed length becomes 2048. So, first
+				// construct the url for /shares using tenant domain but to be on safer side, fallback to what we were using before.
+				try {
+					redeemUrl = `${
```

Comment:
Let's add a config or feature flag to disable this flow. In case something goes wrong or if this causes any regression, it can be disabled by applications / hosts without having to make a code change which could take a long time to propagate.